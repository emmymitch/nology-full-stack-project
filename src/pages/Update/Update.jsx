import "./Update.scss";
import TextInput from "../../components/TextInput/TextInput";
import Button from "../../components/Button/Button";
import { useState } from "react";


const Update = () => {
    const initialData = {
        createdBy: "",
        englishName: "",
        greekName: "",
        majorDomains: [],
        identifiers: [],
        description: ""
    }

    const [formSubmitted, setFormSubmitted] = useState(false);
    const [formData, setFormData] = useState(initialData);

    const handleChange = (event) => {
        switch(event.target.name){
            case "createdBy":
                setFormData({...formData, createdBy: event.target.value});
                break;
            case "englishName": 
                setFormData({...formData, englishName: event.target.value});
                break;
            case "greekName": 
                setFormData({...formData, greekName: event.target.value});
                break;
            case "description": 
                setFormData({...formData, description: event.target.value});
                break;
            case "majorDomains":
                setFormData({...formData, majorDomains: [...formData.majorDomains, event.target.value]});
                break;
            case "identifiers":
                setFormData({...formData, identifiers: [...formData.identifiers, event.target.value]});
                break;
            default:
                break;
        }
    }

    const handleMultiInputChange = (event) => {
        console.log(formData);
    }

    //what the fuck does submitting a form do??? where does the data go??????????
    const handleSubmit = (event) => {
        event.preventDefault();
        console.log(event.target[0])
        setFormSubmitted(true);
        setFormData({...initialData});
    }

    const handleReset = () => {
        setFormData({...initialData});
        setFormSubmitted(false);
    }

    const submitMessage = <h4 className="update__subheading">Entry updated!</h4>
    const pageJSX = 
        <form className="update__form" onSubmit={handleSubmit} >
            <label className="update__form--label" htmlFor="createdBy">Author:</label>
            <TextInput name={"createdBy"} onBlur={handleChange} placeholder={"Who's authoring this entry?"} />
            <br></br>

            <label className="update__form--label" htmlFor="englishName">English Name:</label>
            <TextInput name={"englishName"} onBlur={handleChange} />
            <br></br>

            <label className="update__form--label" htmlFor="greekName">Greek Name:</label>
            <TextInput name={"greekName"} placeholder={"Use Greek characters only"} onBlur={handleChange} />
            <br></br>

            <label className="update__form--label" htmlFor="majorDomains">Major Domains:</label>
            <div className="update__form--domain-list">
                <TextInput name={"majorDomains"} onBlur={handleChange} />
                <TextInput name={"majorDomains"} onBlur={handleChange} />
                <TextInput name={"majorDomains"} onBlur={handleChange} />
                <TextInput name={"majorDomains"} onBlur={handleChange} />
                {/* Remember to change the sneaky 'check form status' function!! */}
                <TextInput name={"majorDomains"} onBlur={handleMultiInputChange} />
            </div>
            <br></br>

            <label className="update__form--label" htmlFor="identifiers">Identifiers:</label>
            <div className="update__form--identifier-list">
                <TextInput name={"identifiers"} placeholder={"e.g. lightning bolts and eagles for Zeus"} onBlur={handleChange} />
                <TextInput name={"identifiers"} placeholder={"e.g. owls, olives, Aegis for Athena"} onBlur={handleChange} />
                <TextInput name={"identifiers"} placeholder={"e.g. lyre, bow, laurel for Apollo"} onBlur={handleChange} />
                <TextInput name={"identifiers"} placeholder={"e.g. a caduceus and winged sandals for Hermes"} onBlur={handleChange} />
                <TextInput name={"identifiers"} placeholder={"e.g. doves and roses for Aphrodite"} onBlur={handleMultiInputChange} />
            </div>
            <br></br>

            <label className="update__form--label" htmlFor="description">Description:</label>
            <textarea 
                className="update__form--description" 
                name="description" 
                placeholder="A short description of the god. Exclude personal opinions."
                onBlur={handleChange}
            />
            <br></br>

            <Button type={"submit"} text={"Submit"} />
        </form>
    ;


    return (
        <div className="update">
            <h1 className="update__title">Add/Update a God</h1>
            <h3 className="update__subheading">Is some information wrong, or do you want to add a new entry? Use this form to do so!</h3>
            <Button className={"button--reset"} type={"reset"} onClick={handleReset} text={"Reset"} />
            <br></br>
            {formSubmitted ? submitMessage : pageJSX}
        </div>
    )
}

export default Update;