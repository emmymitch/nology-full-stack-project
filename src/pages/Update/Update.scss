@use "../../styles/variables" as *;

.update{
    display: flex;
    flex-direction: column;

    & > * {
        margin: 1rem 0;
    }

    &__form{
        display: flex;
        flex-direction: column;
        align-items: center;

        &--label, &--submit, &--reset{
            margin: 0.75rem 0 0.25rem 0;
        }

        &--domain-list, &--identifier-list{
            display: flex;
            flex-direction: column;
        }

        &--description{
            width: 20rem;
            height: 5rem;
            font-size: 0.85rem;
        }
    }
}