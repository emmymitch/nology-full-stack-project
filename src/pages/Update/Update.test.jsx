import Update from "./Update";
import { render } from "@testing-library/react";

it("Should render the Update page", () => {
  const container = render(<Update />)

  expect(container).toMatchSnapshot();
});