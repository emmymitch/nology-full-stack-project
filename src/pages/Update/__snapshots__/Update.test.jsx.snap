// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Update page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="update"
      >
        <h1>
          Update
        </h1>
        <h3
          class="update__subtitle"
        >
          Seen something wrong? Correct it here!
        </h3>
        <br />
        <form
          class="update__form"
        >
          <div
            class="update__form--style-div"
          >
            <div
              class="update__form--input-div"
            >
              <label
                class="update__form--label"
                for="entry-id"
              >
                Card ID:
              </label>
              <input
                class="text-input "
                name="entry-id"
                placeholder="Find the unique ID at the bottom of each card"
                type="text"
                value=""
              />
            </div>
            <br />
            <div
              class="update__form--input-div"
            >
              <label
                class="update__form--label"
                for="entry-categories"
              >
                Category to change:
              </label>
              <select
                class="update__form--dropdown"
                name="entry-categories"
              >
                <option
                  value="englishName"
                >
                  English Name
                </option>
                <option
                  value="greekName"
                >
                  Greek Name
                </option>
                <option
                  value="majorDomains"
                >
                  Major Domains
                </option>
                <option
                  value="identifiers"
                >
                  Identifiers
                </option>
                <option
                  value="description"
                >
                  Description
                </option>
              </select>
            </div>
            <div
              class="update__form--input-div"
            >
              <label
                class="update__form--label"
                for="entry-change"
              >
                New data:
              </label>
              <input
                class="text-input "
                name="entry-change"
                placeholder=""
                type="text"
                value=""
              />
            </div>
          </div>
          <br />
          <p
            class="update__form--request"
          >
            If changing domains or identifiers, please input as domain1,domain2,domain3
          </p>
          <button
            class="button "
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="update"
    >
      <h1>
        Update
      </h1>
      <h3
        class="update__subtitle"
      >
        Seen something wrong? Correct it here!
      </h3>
      <br />
      <form
        class="update__form"
      >
        <div
          class="update__form--style-div"
        >
          <div
            class="update__form--input-div"
          >
            <label
              class="update__form--label"
              for="entry-id"
            >
              Card ID:
            </label>
            <input
              class="text-input "
              name="entry-id"
              placeholder="Find the unique ID at the bottom of each card"
              type="text"
              value=""
            />
          </div>
          <br />
          <div
            class="update__form--input-div"
          >
            <label
              class="update__form--label"
              for="entry-categories"
            >
              Category to change:
            </label>
            <select
              class="update__form--dropdown"
              name="entry-categories"
            >
              <option
                value="englishName"
              >
                English Name
              </option>
              <option
                value="greekName"
              >
                Greek Name
              </option>
              <option
                value="majorDomains"
              >
                Major Domains
              </option>
              <option
                value="identifiers"
              >
                Identifiers
              </option>
              <option
                value="description"
              >
                Description
              </option>
            </select>
          </div>
          <div
            class="update__form--input-div"
          >
            <label
              class="update__form--label"
              for="entry-change"
            >
              New data:
            </label>
            <input
              class="text-input "
              name="entry-change"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
        <br />
        <p
          class="update__form--request"
        >
          If changing domains or identifiers, please input as domain1,domain2,domain3
        </p>
        <button
          class="button "
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
